control/veda_deliver_cielo.py

```diff
@@ -1,186 +1,228 @@
-            '&password=' + self.c24_defaults['c24_pass']
-
-        # Generate Token
-        r1 = requests.get(token_url)
-        if r1.status_code > 299:
-            ErrorObject.print_error(
-                message='Cielo24 API Access Error',
-            )
-            return None
-        api_token = ast.literal_eval(r1.text)["ApiToken"]
-        return api_token
-
-    def listjobs(self):
-        """List Jobs"""
-        api_token = self.tokengenerator()
-        r2 = requests.get(
-            ''.join((
-                self.c24_site,
-                self.c24_joblist,
-                '?v=1&api_token=',
-                api_token
-            ))
+        return cielo24_defaults
+
+    def start_transcription_flow(self):
+        """
+        Start cielo24 transcription flow.
+
+        This will do the following steps:
+        For each preferred language:
+            1. create a new job
+            2. add media url
+            3. perform transcript
+        """
+        job_id = None
+        preferred_langs = self.video_transcript_preferences.get('preferred_langs')
+
+        # update transcript status for video in edx-val
+        VALAPICall(video_proto=None, val_status=None).update_video_transcript_status(
+            self.video.studio_id, VideoTranscriptStatus.IN_PROGRESS, job_id
         )
-        job_list = r2.text
-        return job_list
 
-    def generate_jobs(self, api_token):
+        for perferred_lang in preferred_langs:
+            try:
+                job_id = self.create_job()
+                transcript_process_metadata = TranscriptProcessMetadata.objects.create(
+                    video=self.video,
+                    process_id=job_id,
+                    lang_code=perferred_lang,
+                    provider=TranscriptProvider.CIELO24,
+                    status=TranscriptStatus.IN_PROGRESS
+                )
+                self.create_add_media_url(job_id)
+                self.perform_transcript(job_id, perferred_lang)
+            except Cielo24Error:
+                transcript_process_metadata.status = TranscriptStatus.FAILED
+                transcript_process_metadata.save()
+
+                LOGGER.exception(
+                    '[cielo24] Request failed for video=%s with job_id=%s', self.video.studio_id, job_id
+                )
+                continue
+
+    def perform_transcript(self, job_id, lang_code):
         """
-        'https://api.cielo24.com/job/new?v=1&\
-        api_token=xxx&job_name=xxx&language=en'
+        Request cielo24 to generate transcripts for a video.
         """
-        r3 = requests.get(
+        callback_url = urllib.quote_plus('{}?job_id={}&lang_code={}&org={}&video_id={}'.format(
+            self.callback_base_url,
+            job_id,
+            lang_code,
+            self.video_transcript_preferences.get('org'),
+            self.video.studio_id,
+        ))
+        response = requests.get(
             ''.join((
```

Comment:
we may move this `joining` in a separate class method and use it wher-ever needed.
```
def build_url(base_url, relative_url, **query_params):
    return ''.join(base_url, relative_url, urllib.urlencode(query_params)) 
```